File: 0940-fruit-into-baskets/0940-fruit-into-baskets.java
class Solution {
    public int totalFruit(int[] fruits) {
        Map<Integer,Integer> mp= new HashMap<>();
        int result=0,j=0;
        for(int i=0;i<fruits.length;i++){
            mp.put(fruits[i],mp.getOrDefault(fruits[i],0)+1);
            while(mp.size()==3){
                mp.put(fruits[j],mp.get(fruits[j])-1);
                if(mp.get(fruits[j])==0) mp.remove(fruits[j]);
                j++;
            }
            result=Math.max(result,i-j+1);
        }
        return result;
    }
}

File: 0940-fruit-into-baskets/0940-fruit-into-baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        i,j=0,0
        result=0
        mp={}
        while(i<len(fruits)):
            mp[fruits[i]]=mp.get(fruits[i],0)+1
            while(len(mp)==3):
                mp[fruits[j]]=mp.get(fruits[j])-1
                if(mp[fruits[j]]==0):
                    mp.pop(fruits[j])
                j+=1
            result=max(result,i-j+1)
            i+=1
        return result